src/Features/Lsif/Generator/Graph/SemanticTokensCapabilities.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.CodeAnalysis.LanguageServerIndexFormat.Generator.Graph
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using System.Threading.Tasks;
+    using Newtonsoft.Json;
+
+    internal sealed class SemanticTokensCapabilities
```

Comment:
Should we be using the SemanticTokensLegend type that already exists in LSP for this?

Reply:
I see two possible outcomes over the evolution of this tool:

- We never add any more semantic tokens capabilities - in that case, yes, using SemanticTokensLegend would eliminate one boilerplate class.

- We add more semantic tokens capabilities - these will almost definitely be different than the LSP ones requiring a custom type anyways.

I don't think it's worth spending any more time on this. When we standardize Semantic Tokens in the LSIF spec, this will likely require changes regardless.

Reply:
Can we track this question in the spec somewhere if it's not already? Or does the spec not care and this becomes a issue for the LSP library?

Reply:
Yeah this will be part of the Semantic Tokens LSIF spec proposal. Currently this exists as just a [draft in my document](https://microsoft-my.sharepoint.com/:w:/p/chgund/EaxCHsz6AxpNjCrmbZ4VpLEBlZbcJ__WRfGos8ZInAplEA?e=Vfyz8U).

I'll send a spec addition proposal on the LSIF site once I have finished validating the approach works.